internal/types/array.go

```diff
@@ -128,3 +132,53 @@ func (a *Array) Append(values ...any) error {
 	a.s = append(a.s, values...)
 	return nil
 }
+
+// HasSameTypeElements returns true is Array elements has the same type.
+func (a *Array) HasSameTypeElements() bool {
+	var prev string
+	for _, i := range a.s {
+		var cur string
+		switch i := i.(type) {
+		case int32, int64:
```

Comment:
Let's use the same consistent order as everywhere: https://github.com/FerretDB/FerretDB/blob/6e8a7773a9867f76e982ed9229594d6c899e8c46/internal/types/types.go#L26-L44

Reply:
Fixed